### ISSUE TITLE
Adding a feature to toggle the TOC when viewport is small (i.e. mobile view)

### ISSUE THREAD

**Issue by colbygogan** [observablehq/framework#1136]
Currently when the viewport is resized to a smaller width, the TOC will disappear.  It would be helpful to still be able to see the TOC when the viewport is shrunk and/or implement a toggle to show the TOC - similar to the image attached.

![Capture d’écran 2024-03-23 à 16 56 43](https://github.com/observablehq/framework/assets/77361345/acd86d5d-c3f2-4dad-81a6-91fab7793a5a)

**Comment by mbostock** [comment]
Yes, I’d like this. VitePress does this well by inserting a position-fixed header and then using a popover for the contents.

<img width="612" alt="Screenshot 2024-05-08 at 5 54 02 PM" src="https://github.com/observablehq/framework/assets/230541/7cf26b29-b804-4b59-8261-bad0193a01bf">
